### DISCUSSION TITLE
The tagged releases on GitHub are different to those available for download on the main website

### DISCUSSION THREAD

**Discussion by matty-r** [apache/netbeans#6651] [Q&A] [answered]
Hi all,

I'm not talking about the hash, but the actual contents between the zips are different. I'm looking to fix up the Arch Linux Netbeans package, and i've noticed that the GitHub version of the source contains the Rust directory whereas downloading from the website (for example https://www.apache.org/dyn/closer.lua/netbeans/netbeans/19/netbeans-19-source.zip) does not. And the structure of the zip is different.

Is there a reason for this? If you try and build with the one downloaded from the website it will complain that the Rust directory is missing.

**Comment by neilcsmith-net** [comment, marked as answer]
The zip on the tag is the automatic GitHub generated snapshot of the source tree at that hash.

The official release is the `netbeans-19-source.zip` one from apache.org.

The source release pre-bundles the release info files that are downloaded when building from a source checkout.

The source release does not include all clusters (eg. Rust), as that has not been designated as releasable so far.  This allows development to happen in the main source tree without affecting releases.

The short answer - please prefer the official signed and released source zip from the Apache site for packaging NetBeans. Thanks!

**Reply by neilcsmith-net** [reply]
> If you try and build with the one downloaded from the website it will complain that the Rust directory is missing.

To add, only if you try and specify an alternative (eg. `full`) cluster config.  Leave the default, or use a subset of it.

**Reply by mbien** [reply]
> The zip on the tag is the automatic GitHub generated snapshot of the source tree at that hash.

yeah. There is unfortunately no way to turn this off at the moment. The zip attached to the github release list is simply a snapshot of the whole repository, which is not the same as the source bundles on the download page which fit to the release.

Rust is still baking and is not part of the release yet.

**Reply by matty-r** [reply]
Thanks for the response - that makes a lot of sense. The only issue is if you set to do a full build based on the Apache.org site, it tries to build with Rust and will fail.

**Reply by neilcsmith-net** [reply]
@matty-r use `release` not `full` although that is default anyway. We should consider renaming full to something like all-plus-unreleased! 😄 

Incidentally, if you do include the Rust cluster in your package, please be clear it is _not_ a NetBeans release!

**Reply by matty-r** [reply]
@neilcsmith-net ahh ok, yea I switched it to release and appeared to fix it up. Maybe `dev` build would be more appropriate?

Anyway, thanks heaps for your help. Really appreciate it.
